### ISSUE TITLE
[Error: Mass is not valid]

### ISSUE THREAD

**Issue by davidarnaub7** [matinzd/react-native-health-connect#61]
I'm trying to add a complete recipe macros to Nutrition section.

```
{
"dietaryFiber": {"unit": "grams", "value": 13}, "endTime": "2024-01-23T08:18:03.000Z", 
"energy": {"unit": "kilocalories", "value": 760},
 "mealType": 1, 
"protein": {"unit": "grams", "value": 38},
 "recordType": "Nutrition", 
"saturatedFat": {"unit": "grams", "value": 7}, 
"sodium": {"unit": "grams", "value": 1},
 "startTime": "2024-01-23T08:18:00.000Z",
 "sugar": {"unit": "grams", "value": 39}, 
"totalCarbohydrate": {"unit": "grams", "value": 77},
 "totalFat": {"unit": "grams", "value": 34}
}
```

This is the complete record object. Typescript not complaining and everything seems fine but I'm getting this error.

Any ideas?

**Comment by davidarnaub7** [comment]
@matinzd did you see something?? Thanks!

**Comment by matinzd** [comment]
Yes. I will send a fix alongside the permission issue on Android 14. You may need to wait for some time.

**Comment by matinzd** [comment]
Fixed in [v2.0.0](https://github.com/matinzd/react-native-health-connect/releases/tag/v2.0.0).
